posthog/temporal/llm_analytics/run_evaluation.py

```diff
@@ -45,12 +46,15 @@ async def fetch_target_event_activity(inputs: RunEvaluationInputs, team_id: int)
             distinct_id,
             person_id
         FROM events
-        WHERE uuid = %(event_id)s AND team_id = %(team_id)s
+        WHERE team_id = %(team_id)s
+            AND toDate(timestamp) = toDate(parseDateTimeBestEffort(%(target_timestamp)s))
+            AND event = '$ai_generation'
+            AND uuid = %(event_id)s
```

Comment:
Not sure if **all** events have one, but if you put a `distinct_id` as well, it will fully use the sorting key:
```
team_id, toDate(timestamp), event, cityHash64(distinct_id), cityHash64(uuid)
```